src/Concordium/Client/Runner.hs

```diff
@@ -1507,32 +1566,28 @@ processAccountCmd action baseCfgDir verbose backend =
         Nothing -> do
-          logInfo [[i|the ACCOUNT argument was not provided; using the default account name '#{defaultAccountName}'|]]
-          return defaultAccountName
+          case Map.lookup defaultAccountName (bcAccountNameMap baseCfg) of
+            Nothing -> do
+              logFatal [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used, but no account with that name exists.|]]
+            Just acc -> do
+              logInfo [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used.|]]
+              return . Text.pack $ show acc
         Just acc -> return acc
 
-      accountIdentifier <- case Types.decodeAccountIdentifier (Text.encodeUtf8 input) of
-                Just _ -> return input -- input is a wellformed account identifier
-                Nothing -> case Map.lookup input (bcAccountNameMap baseCfg) of
-                    Just a -> return $ Text.pack $ show a -- input is the local name of an account
-                    Nothing -> if isNothing inputMaybe
-                               then logFatal [[i|The ACCOUNT argument was not provided; so the default account name '#{defaultAccountName}' was used, but no account with that name exists.|]]
-                               else logFatal [[i|The identifier '#{input}' is neither a credential registration ID, an account index, the address nor the name of an account|]]
+      accountIdentifier <-
+        case Types.decodeAccountIdentifier (Text.encodeUtf8 input) of
+          Just v -> return v -- input is a wellformed account identifier
+          Nothing -> logFatal [[i|The identifier '#{input}' is neither a credential registration ID, an account index, the address nor the name of an account|]]
 
       (accInfo, na, dec) <- withClient backend $ do
         -- query account
-        (bbh, accInfoValue :: GRPCResult Value) <- withBestBlockHash block (\bbh -> (bbh,) <$> getAccountInfo accountIdentifier bbh)
-        accInfo <- do 
-          case accInfoValue of
-            Left err -> logFatal [err]
-            Right aiv -> case AE.fromJSON $ grpcResponseVal aiv of
-                            AE.Error err -> logFatal ["Cannot decode account info response from the node: " ++ err]
-                            AE.Success Nothing -> logFatal [printf "Account %s does not exist on the chain." $ show accountIdentifier]
-                            AE.Success (Just air) -> return air
+        bhInput <- readBlockHashOrDefault Best block
+        accInfo <- getResponseValueOrDie =<< getAccountInfoV2 accountIdentifier bhInput
         -- derive the address of the account from the the initial credential
-        resolvedAddress <- case Map.lookup (ID.CredentialIndex 0) (Types.aiAccountCredentials accInfo) of
-                            Nothing -> logFatal [printf "No initial credential found for the account identified by '%s'" accountIdentifier ]
```

Comment:
How come `show` works here, when `AccountIdentifier` is not an instance of `show`?

I even get `No instance for (Show Types.AccountIdentifier)` when I try now? What am I overlooking?

Reply:
As I understand it, `printf` doesn't use `Show` but rather `PrintfArg`. Also, it is not typesafe, so it is better to use the string interpolation with `logFatal [[i|No initial credential found for the account identified by '#{accountIdentifier}'|]]`.

Reply:
Thanks. I'm not sure I understand how that works, but in any case, using string interpolation was what I wanted. I'll add the `Show` instance then.